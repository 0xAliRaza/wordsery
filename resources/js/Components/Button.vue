<template>
    <button :type="type">
        <slot />
    </button>
</template>

<script>
export default {
    props: {
        type: {
            type: String,
            default: 'submit',
        },
    }
}
</script>
<style lang="scss" scoped>
@import '@scss/abstracts';
button {
    @include secondary-font;
    font-size: pxToRem(15);
    font-weight: 500;
    border: none;
    padding: pxToRem(8) pxToRem(16);
    border-radius: 100px;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    background: $primary;
    color: $light;
    cursor: pointer;
    text-transform: uppercase;
    transition: all 0.2s ease;
    &:hover, &:focus {
        outline: none;
        background: $primary-dark;
    }
}
</style>
