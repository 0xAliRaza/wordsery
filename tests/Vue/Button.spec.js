import Button from '@/Components/Button.vue'
import {mount} from '@vue/test-utils'
describe("Button", () => {
    it("renders correctly", () => {
        const wrapper = mount(Button);
        expect(wrapper.html()).toMatchSnapshot();
    });
});