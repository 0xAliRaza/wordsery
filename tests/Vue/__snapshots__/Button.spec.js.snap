// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `"<button type=\\"submit\\" class=\\"\\"></button>"`;

exports[`Foo renders a div 1`] = `"<button type=\\"submit\\" class=\\"\\"></button>"`;
